presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -441,9 +442,12 @@ public boolean isSupportedPartitionKey()
             case DOUBLE:
             case INET:
             case INT:
+            case SMALLINT:
+            case TINYINT:
             case FLOAT:
             case DECIMAL:
             case TIMESTAMP:
+            case DATE:
```

Comment:
i tend to put date before timestamp ("simpler one first")

also, would be nice to put all the inteer types together (TINYINT, SMALLINT, INT, BIGINT)